### PR TITLE
HPCC-29353 Combine init containers performing chown

### PR DESCRIPTION
**Pull request by jakesmith**
Create a single initContainer that is responsible for any 'forcePermission' chown'ing.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: helm/hpcc/templates/_helpers.tpl [hunk 1]
```diff
@@ -761,7 +761,15 @@ A kludge to ensure mounted storage (e.g. for nfs, minikube or docker for desktop
 # This is only required when mounting a remote filing systems from another container or machine.
 # NB: this includes where the filing system is on the containers host machine .
 # Examples include, minikube, docker for desktop, or NFS mounted storage.
-{{- $permCmd := printf "chown -R %v:%v %s || true" .uid .gid .volumePath }}
+{{- $permCmd := "" -}}
+{{- $uid := .uid -}}
+{{- $gid := .gid -}}
+{{- range $index, $volume := .volumes }}
+ {{- if ne $index 0 }}
+  {{- $permCmd = printf "%s & " $permCmd -}}
```
Comment (ghalliday) [inline helm/hpcc/templates/_helpers.tpl:769]: Could this container terminate if the later chown succeeds, but the first are still running?

**Review by ghalliday — COMMENTED** [review]
@jakesmith one question

File: helm/hpcc/templates/_helpers.tpl (hunk 1 above)
Comment (jakesmith) [reply helm/hpcc/templates/_helpers.tpl:769]: good question, I think not, I test with earlier chown's failing and future ones succeeding,
but not laters failing/succeeding/ while earlier still running.

I'll circle back and check when I get back.

**Review by richardkchapman — APPROVED** [review]
I did not spot any issues. I agree with Gavin that planesToChown would be a better name

File: helm/hpcc/templates/_helpers.tpl [hunk 2]
```diff
@@ -788,19 +798,27 @@ NB: uid=10000 and gid=10001 are the uid/gid of the hpcc user, built into platfor
 {{- $planes := ($storage.planes | default list) -}}
 {{- $includeCategories := .includeCategories | default list -}}
 {{- $includeNames := .includeNames | default list -}}
+{{- $planesToDo := list -}}
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply helm/hpcc/templates/_helpers.tpl]: changed.

**Comment by jakesmith** [comment]
@ghalliday - please see new commit.

**Review by ghalliday — APPROVED** [review]
@jakesmith please squash.  Is 9.0.x the correct target (I was wondering if needs to be earlier?)

**Pull request merged**